cpp/benchmarks/reduction/scan_structs.cpp

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <benchmarks/common/generate_input.hpp>
+#include <benchmarks/fixture/rmm_pool_raii.hpp>
+
+#include <nvbench/nvbench.cuh>
+
+#include <cudf/column/column_factories.hpp>
+#include <cudf/detail/scan.hpp>
+
+static constexpr cudf::size_type num_struct_members = 8;
+static constexpr cudf::size_type max_int            = 100;
+static constexpr cudf::size_type max_str_length     = 32;
+
+static void nvbench_structs_scan(nvbench::state& state)
+{
+  cudf::rmm_pool_raii pool_raii;
+
+  auto const null_probability = [&] {
+    auto const null_prob_val = state.get_float64("null_probability");
+    return null_prob_val > 0 ? std::optional{null_prob_val} : std::nullopt;
+  }();
+  auto const size    = static_cast<cudf::size_type>(state.get_int64("data_size"));
+  auto const profile = static_cast<data_profile>(
+    data_profile_builder()
+      .null_probability(null_probability)
+      .distribution(cudf::type_id::INT32, distribution_id::UNIFORM, 0, max_int)
+      .distribution(cudf::type_id::STRING, distribution_id::NORMAL, 0, max_str_length));
+
+  auto data_table = create_random_table(
+    cycle_dtypes({cudf::type_id::INT32, cudf::type_id::STRING}, num_struct_members),
+    row_count{size},
+    profile);
+  auto [null_mask, null_count] = create_random_null_mask(size, null_probability);
+  auto const input             = cudf::make_structs_column(
+    size, std::move(data_table->release()), null_count, std::move(null_mask));
+
+  auto const agg         = cudf::make_min_aggregation<cudf::scan_aggregation>();
+  auto const null_policy = static_cast<cudf::null_policy>(state.get_int64("null_policy"));
+
+  state.exec(nvbench::exec_tag::sync, [&](nvbench::launch& launch) {
```

Comment:
Recommend setting the stream for the nvbench state:
```
state.set_cuda_stream(nvbench::make_cuda_stream_view(cudf::get_default_stream().value()));
```
Also, it would be good if it is possible to provide some throughput metrics.
Example: https://github.com/rapidsai/cudf/blob/ec45c18c0f88acbf8f518a3ac87c3181da93ee2c/cpp/benchmarks/string/like.cpp#L87-L89
(The metrics are not a requirement)

Reply:
Thanks. The metrics are not trivial to compute in this case, as the input here is a complex structs column.